packages/ember-testing/lib/helpers.js

```diff
@@ -59,14 +59,13 @@ function visit(app, url) {
   var router = app.__container__.lookup('router:main');
   app.boot().then(function() {
     router.location.setURL(url);
+    run(app.__deprecatedInstance__, 'handleURL', url);
```

Comment:
Lets tweak this slightly so it only runs in the `else` case below (via a guard or something).  Otherwise, we end up double routing on the first `visit` call (a default app boot would have transitioned into the initial URL then we call `handleURL` to transition again).